src/backend/xilinx/memory_axi.rs

```diff
@@ -98,11 +100,11 @@ impl MemoryInterface for AxiInterface {
         name: &str,
         bus_data_width: u64,
         bus_addr_width: u64,
-        // address_width: u64,
         data_width: u64,
+        memory_size: u64,
+        addr_width: u64,
     ) -> v::Module {
         let mut module = v::Module::new(name);
-        let memory_size = 32;
         let memory_size_bits: u64 = utils::math::bits_needed_for(memory_size); // TODO make memory size parametric
```

Comment:
Hopefully `memory_size_bits == addr_width`… maybe adding an `assert` here to check that would be cool?

Reply:
This is actually slightly changed in #1103.
One thing that occurred to me on the subject is that because calyx-memory has the width of it's address port defined through an argument, there could be a (perhaps strange? Not sure how common this would be) case where there is a wider address port than needed for the size of a memory. In that case the two wouldn't be equal.